pkg/sql/job_statistics.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/retry"
+)
+
+type metricsPoller struct {
+	job *jobs.Job
+}
+
+var _ jobs.Resumer = &metricsPoller{}
+
+var metricsPollerRetryOptions = retry.Options{
+	InitialBackoff: 5 * time.Millisecond,
+	Multiplier:     2,
+	MaxBackoff:     10 * time.Second,
+}
+
+// OnFailOrCancel is a part of the Resumer interface.
+func (mp *metricsPoller) OnFailOrCancel(
+	ctx context.Context, execCtx interface{}, jobErr error,
+) error {
+	if jobErr != nil {
+		log.Errorf(ctx, "failed polled stats job: %v", jobErr)
```

Comment:
Since this job is meant to always run, wouldn't we always emit this message? Because even when node shuts down, jobErr would probably be some sort of context cancelled or whatnot?

Reply:
Good point. I don't think it's worth having the log message.